File: DSA/Aarohi-Sharma/Day35_q2.cpp
class Solution {
public:
    int consecutiveNumbersSum(int n) {
        int res = 1, count;
        while (n % 2 == 0) 
            n /= 2;
        for (int i = 3; i <= n; res *= count + 1, i += 2)
            for (count = 0; n % i == 0; n /= i, count++);
        return n == 1 ? res : res * 2;
    }
};

File: DSA/Ishita-Nigam/Day35/Day35_Q2.java
import java.util.Scanner;

public class Day35_Q2 {
    public int consecutiveNumbersSum(int n) {
        int count = 0;
        for (int i = 2; i < n; i++) {
            int sum_1 = i * (i + 1) / 2;
            if (sum_1 > n)
                break;
            if ((n - sum_1) % i == 0)
                count++;
        }
        return count + 1;
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Enter a number: ");
        int n = scanner.nextInt();
        scanner.close();

        Day35_Q2 solution = new Day35_Q2();
        System.out.println("Number of ways to write " + n + " as a sum of consecutive numbers: " + solution.consecutiveNumbersSum(n));
    }
}
